app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,67 @@
+class AlbumsController < ApplicationController
+  def index
+    @albums = Album.order(ranking: :desc)
+  end
+
+  def show
+    id = params[:id]
+    @album = Album.find(id)
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def upvote
+    @album = Album.find(params[:id])
+    if @album.ranking.nil?
+      ranking = 1
+    else
+      ranking = @album.ranking + 1
+    end
+    @album.update(ranking: ranking)
+    redirect_to album_path(@album.id)
+  end
+
+  def create
+    @album = Album.create(album_params[:album])
```

Comment:
So if you've added @album.ranking = 1, you wouldn't have to use ``if statement`` in upvote method.